### PR TITLE
#11 - Added Per Page

### PR DESCRIPTION
**Pull request by dmitrii-casapiuc**

### CODE REVIEW

File: node/src/shared/store/actions/products.js [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+import {FETCH_PRODUCT_SUCCESS, FETCH_PRODUCTS_SUCCESS, PRODUCTS_PAGINATION} from "./actionsTypes"
+import axios from "axios"
 
-export function fetchProducts() {
+export function fetchProducts(per_page = '', page = '') {
     return async dispatch => {
+        let params = ''
+
+        if (per_page !== '' && page !== '') {
+            params = `/?per_page=${per_page}&page=${page}`
```
Comment (WictorT) [inline node/src/shared/store/actions/products.js]: use methods for query params

File: node/src/shared/store/actions/products.js [hunk 2]
```diff
@@ -1,11 +1,19 @@
-import {FETCH_PRODUCTS_SUCCESS, FETCH_PRODUCT_SUCCESS} from "./actionsTypes"
-import axios from 'axios'
+import {FETCH_PRODUCT_SUCCESS, FETCH_PRODUCTS_SUCCESS, PRODUCTS_PAGINATION} from "./actionsTypes"
+import axios from "axios"
 
-export function fetchProducts() {
+export function fetchProducts(per_page = '', page = '') {
```
Comment (WictorT) [inline node/src/shared/store/actions/products.js]: refactor to camelCase

File: node/src/app/HomePage/HomePage.js [hunk 3]
```diff
@@ -1,108 +1,209 @@
...
                                     ))}
                                 </Grid>
+                                {renderPageNumbers}
+                                <InputLabel htmlFor="age-customized-select"></InputLabel>
+                                <Select
+                                    value={perPage}
+                                    onChange={this.handleChange}
+                                    input={<BootstrapInput name="age" id="age-customized-select"/>}
+                                >
+                                    <MenuItem value={1}>1</MenuItem>
```
Comment (WictorT) [inline node/src/app/HomePage/HomePage.js]: Extract options to constants

File: node/src/app/HomePage/HomePage.js [hunk 4]
```diff
@@ -1,108 +1,209 @@
...
+    },
+})
 
 class HomePage extends Component {
 
     componentDidMount() {
         this.props.fetchProducts()
     }
 
+    handlePage = (event, perPage) => {
```
Comment (WictorT) [inline node/src/app/HomePage/HomePage.js]: Use a single method for bboth

File: node/src/app/HomePage/HomePage.js [hunk 5]
```diff
@@ -1,108 +1,209 @@
...
+                                                        onClick={() => history.push("/" + product.id)}
+                                                    >
                                                         View
                                                     </Button>
                                                 </CardActions>
                                             </Card>
                                         </Grid>
                                     ))}
                                 </Grid>
+                                {renderPageNumbers}
```
Comment (WictorT) [inline node/src/app/HomePage/HomePage.js]: redundant extraction here. Put code right here.

File: node/src/app/HomePage/HomePage.js [hunk 6]
```diff
@@ -1,108 +1,209 @@
...
+                </Link>
+            )
+        })
+
         return (
             <React.Fragment>
                 <main>
                     {
-                        loading
+                        loader
```
Comment (WictorT) [inline node/src/app/HomePage/HomePage.js]: `loading` was correct

File: node/src/app/HomePage/HomePage.js [hunk 7]
```diff
@@ -1,108 +1,209 @@
...
+            loader,
+            totalPages,
+            perPage
+        } = this.props
+
+        const pageNumbers = []
 
-        if ( ! products ) {
-            loading = true
+        for (let i = 1; i <= totalPages; i++) {
```
Comment (WictorT) [inline node/src/app/HomePage/HomePage.js]: use `range` here: https://lodash.com/docs/4.17.11#range

**Review by WictorT — CHANGES_REQUESTED** [review]

File: node/src/app/HomePage/HomePage.js (hunk 7 above)
Comment (dmitrii-casapiuc) [reply node/src/app/HomePage/HomePage.js]: Done

File: node/src/app/HomePage/HomePage.js (hunk 5 above)
Comment (dmitrii-casapiuc) [reply node/src/app/HomePage/HomePage.js]: Done

File: node/src/app/HomePage/HomePage.js (hunk 6 above)
Comment (dmitrii-casapiuc) [reply node/src/app/HomePage/HomePage.js]: Done

File: node/src/app/HomePage/HomePage.js (hunk 3 above)
Comment (dmitrii-casapiuc) [reply node/src/app/HomePage/HomePage.js]: Done

File: node/src/shared/store/actions/products.js (hunk 2 above)
Comment (dmitrii-casapiuc) [reply node/src/shared/store/actions/products.js]: Done

File: node/src/shared/store/actions/products.js (hunk 1 above)
Comment (dmitrii-casapiuc) [reply node/src/shared/store/actions/products.js]: Done

File: node/src/app/Products/Products.js [hunk 8]
```diff
@@ -1,6 +1,7 @@
 import React, {Component} from "react"
 import {connect} from "react-redux"
 import classNames from "classnames"
+import _ from 'lodash'
```
Comment (WictorT) [inline node/src/app/Products/Products.js]: use double quotes `""` as everywhere

File: node/src/app/Products/Products.js [hunk 9]
```diff
@@ -91,55 +92,51 @@ const styles = theme => ({
     },
 })
 
-class HomePage extends Component {
+const optionsChangePerPage = [1, 3, 5]
```
Comment (WictorT) [inline node/src/app/Products/Products.js]: `perPageOptions`

File: node/src/app/Products/Products.js [hunk 10]
```diff
@@ -173,16 +170,40 @@ class HomePage extends Component {
                                         </Grid>
                                     ))}
                                 </Grid>
-                                {renderPageNumbers}
+                                {
+                                    _.range(1, totalPages + 1).map(number => {
```
Comment (WictorT) [inline node/src/app/Products/Products.js]: renamer `number` to a more explicit name. ex. `page` or `pageNumber`.

File: node/.eslintrc.json [hunk 11]
```diff
@@ -0,0 +1,3 @@
+{
+  "extends": "react-app"
```
Comment (WictorT) [inline node/.eslintrc.json:2]: gggg

Comment (WictorT) [reply node/.eslintrc.json:2]: rrr

File: node/src/app/Products/Products.js (hunk 8 above)
Comment (dmitrii-casapiuc) [reply node/src/app/Products/Products.js]: Done

File: node/src/app/Products/Products.js (hunk 9 above)
Comment (dmitrii-casapiuc) [reply node/src/app/Products/Products.js]: Done

File: node/src/app/Products/Products.js (hunk 10 above)
Comment (dmitrii-casapiuc) [reply node/src/app/Products/Products.js]: Done

**Pull request merged**